Wemo insight: additional parameters added #3538

toby200 opened the pull request:
Fix for #3538 -- added additional parameters for wemo insight plug

More info in the [forum thread] (https://community.openhab.org/t/wemo-insight-plug-adding-new-params-to-binding/4969) with @hmerk

teichsta commented:
could you have a look into this PR please @hmerk?

hmerk commented:
@teichsta, jes, for shure.
Will review end of the week, cause I'm on a business trip actually.

Best
Hans-Jörg

teichsta commented:
perfect, thanks!

hmerk commented:
@toby200, thanks for your contribution. Could you also update the WIKI with your addition, marked as available with version 1.8.

@teichsta LGTM, please feel free to merge for 1.8 release.

Best
Hans-Jörg

toby200 commented:
Thanks, [wiki page](https://github.com/openhab/openhab/wiki/Wemo-Binding) is updated.

Will likely be next week before I get openhab2 version done as I'm away a few days and I couldn't work out how to build the jars yet.

hmerk commented:
@toby200 
Just found the last unknown parameter : It's the standBy Limit value
Could you add this as well, as I already prepared for ESH Binding.

Thanks
Hans-Jörg

toby200 commented:
Great - will look at this over the weekend.

teichsta commented:
ok, so i'll wait for this addition before merging this PR, right?

toby200 commented:
Yes please

toby200 commented:
@hmerk @teichsta 
standbyLimit param is now added too. I've also updated the README.md

Thanks,
Toby

hmerk commented:
@toby200 great, thanks!
@teichsta  LGTM

Best
Hans-Jörg

toby200 commented:
I neglected to set a rounding mode on the BigDecimal.divide() calls, which causes issues with certain specific values - will fix shortly so please hold off on the merge still

teichsta commented on bundles/binding/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/WemoBinding.java:
```diff
@@ -196,6 +270,7 @@ protected void execute() {
 
 				} catch (Exception e) {
 					e.printStackTrace();
```
remove printStackTrace

teichsta commented:
Thanks for incorporating the feedback @toby200 and sorry for stumbling about on last thing (remove printStackTrace()). Could you please remove this line? Thanks, Thomas E.-E.

toby200 commented:
No problem at all - I did wonder about it myself but wasn't sure and was hesitant to remove someone else's code. I've updated and squashed the commit, hope it looks ok now.

Many thanks, 
Toby

teichsta commented:
great, thanks! Could you or @hmerk please finally update the wiki page accordingly?

toby200 commented:
All done :)

teichsta commented:
great thanks! Looking forward to many more contributions from you, sir :+1:
